### PR TITLE
Multipart semantics

### PR DESCRIPTION
**Pull request by cabo**
Address Benjamin Kaduk's outstanding DISCUSS.

### CODE REVIEW

File: draft-ietf-core-multipart-ct.md [hunk 1]
```diff
@@ -87,17 +99,24 @@ substituting a null value for the representation of an optional part,
 which indicates that the part is not present.
 
 A third situation that is common only ever has a single representation
-in the sequence, which selects one of a set of formats possible for
-this situation.  This kind
-of union of formats may also make the presence of the actual
-representation optional, the omission of which leads to a zero-length
-array.
+in the sequence, where the sender selects just one of a set of formats
```
Comment (thomas-fossati) [inline draft-ietf-core-multipart-ct.md]: Isn't this the multipart/alternative semantics?

File: draft-ietf-core-multipart-ct.md [hunk 2]
```diff
@@ -87,17 +99,24 @@ substituting a null value for the representation of an optional part,
...
 Where these rules are not sufficient for an application, it might
 still use the general format defined here, but register a new media
 type and an associated Content-Format identifier to associate the
 representation with these more specific semantics instead of using
-application/multipart-core.
+the application/multipart-core media type.
+
+Also, future specifications might want to define rough equivalents for
+other multipart media types with specific semantics not covered by the
+present specification, such as multipart/alternative (Section 5.1.4 of
```
Comment (thomas-fossati) [inline draft-ietf-core-multipart-ct.md]: See comment above.

File: draft-ietf-core-multipart-ct.md [hunk 3]
```diff
@@ -72,10 +73,21 @@ This simple and efficient binary framing mechanism can be employed to
...
 multiple already existing media types.
 
+As the name of the media-type suggests, it is inspired by the
+multipart media types that started to be defined with the original set
+of MIME specifications {{-mime-types}}.  However, while those needed
+to focus on the syntactic aspects of integrating multiple
+representations into one e-mail, protocols providing full data
+transparency such as CoAP as well as readily available encoding
+formats such as CBOR {{-cbor}} shift the focus towards the intended
+use of the representations combined.  In this respect, the basic
```
Comment (thomas-fossati) [inline draft-ietf-core-multipart-ct.md]: s/representations combined/combined representations/ ?

**Comment by cabo** [comment]
On Aug 19, 2019, at 18:50, Thomas Fossati <notifications@github.com> wrote:
> 
> +in the sequence, where the sender selects just one of a set of formats
> 
> Isn’t this the multipart/alternative semantics?

No: with that, the sender sends along all formats and the recipient chooses based on its capabilities or preferences.  Appropriate for single-shot mail, not at all appropriate for a request/response situation (at least not for the response part).

https://tools.ietf.org/html/rfc2046#section-5.1.4

Grüße, Carsten

**Comment by thomas-fossati** [comment]
> On Aug 19, 2019, at 18:50, Thomas Fossati ***@***.***> wrote: +in the sequence, where the sender selects just one of a set of formats Isn’t this the multipart/alternative semantics?
> No: with that, the sender sends along all formats and the recipient chooses based on its capabilities or preferences. Appropriate for single-shot mail, not at all appropriate for a request/response situation (at least not for the response part). https://tools.ietf.org/html/rfc2046#section-5.1.4 Grüße, Carsten

Ah sorry, hadn't read properly (it's sender, not receiver).

**Comment by thomas-fossati** [comment]
LGTM

**Pull request merged**